scripts/mutect2_wdl/unsupported/mutect2_opt.wdl

```diff
@@ -100,30 +122,48 @@ workflow Mutect2 {
         normal_bam = normal_bam,
         normal_bam_index = normal_bam_index,
         pon = pon,
-        pon_index = pon_index,
         gnomad = gnomad,
-        gnomad_index = gnomad_index,
         output_vcf_name = ProcessOptionalArguments.output_name,
         gatk4_jar_override = gatk4_jar_override,
         preemptible_attempts = preemptible_attempts,
         m2_docker = m2_docker,
         m2_extra_args = m2_extra_args,
-	is_bamOut = is_bamOut,
+        is_bamOut = is_bamOut,
+        disk_space_gb = m2_per_scatter_size,
         auth = auth
     }
+
+    Float sub_vcf_size = size(M2.output_vcf, "GB")
+    Float sub_bamout_size = size(M2.output_bamOut, "GB")
   }
 
+  call SumFloats as SumSubVcfs {
+    input:
+      sizes = sub_vcf_size,
+      preemptible_attempts = preemptible_attempts,
+  }
+
+  Int merge_vcf_size = ceil(SumSubVcfs.total_size * large_input_to_output_mulitplier) + disk_pad
+
   call MergeVCFs {
     input:
-      gatk4_jar = gatk4_jar,
       input_vcfs = M2.output_vcf,
       output_vcf_name = ProcessOptionalArguments.output_name,
       gatk4_jar_override = gatk4_jar_override,
       preemptible_attempts = preemptible_attempts,
+      disk_space_gb = merge_vcf_size,
       m2_docker = m2_docker
   }
 
   if (is_bamOut) {
+    call SumFloats as SumSubBamouts {
```

Comment:
Do we really want to spin up a task just to do calculation of file sizes?  That seems really wasteful...

Reply:
If you have any other way of doing this I'd be happy to change.  There no way to sum values in an array afaik.  See your previous comments from 13 days ago.  Wasteful may be a little strong, it costs like 3 min and $.0001 to run this.  It is definitely not ideal especially with `SumSubVcfs` being on the critical path of the workflow so it actually does add to the wall clock time.  This task should definitely be replaced once its possible to do in wdl but until then I think this is the best we can do unfortunately.